cmd/depdiff/dependencies.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package depdiff
+
+// ChangeType is the change type (added, updated, removed) of a dependency.
+type ChangeType string
+
+const (
+	Added   ChangeType = "added"
+	Updated ChangeType = "updated"
+	Removed ChangeType = "removed"
+)
+
+// IsValid determines if a ChangeType is valid.
+func (ct *ChangeType) IsValid() bool {
+	switch *ct {
+	case Added, Updated, Removed:
+		return true
+	default:
+		return false
+	}
+}
+
+// Dependency is a dependency.
+type Dependency struct {
+	// // IsDirect suggests if the dependency is a direct dependency of a code commit.
+	// TODO: IsDirect remains a future feature since the current GitHub Dependency Review API
+	// mixes up direct and indirect dependencies in manifest files of different ecosystems.
+	IsDirect bool
+
+	// ChangeType indicates whether the dependency is added or removed.
+	ChangeType ChangeType `json:"change_type"`
+
+	// ManifestFileName is the name of the manifest file of the dependency, such as go.mod for Go.
+	ManifestFileName string `json:"manifest"`
+
+	// Ecosystem is the name of the package management system, such as NPM, GO, PYPI.
+	Ecosystem string `json:"ecosystem" bigquery:"System"`
```

Comment:
Do we need the `bigquery` metadata to the definition?

Reply:
For v0, since we don't use any bigquery data sources to fetch data fields, I think the `bigquery` tags are not necessary. Should I remove them in v0, or leave them there for future versions to use?

Reply:
yes, remove everything that's not necessary for this PR. Thanks

Reply:
np, thx